mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/annotation/ElementReferenceTypeAnnotation.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.declaration.type.annotation;
+
+import org.mule.metadata.api.annotation.TypeAnnotation;
+import org.mule.metadata.api.model.ObjectFieldType;
+import org.mule.runtime.api.meta.model.parameter.ElementReference;
+
+/**
+ * A public {@link TypeAnnotation} intended to be used on {@link ObjectFieldType} types in order to
+ * communicate an associated {@link ElementReference}
+ *
+ * @since 1.0
+ */
+public class ElementReferenceTypeAnnotation implements TypeAnnotation {
```

Comment:
This class should be added to `ExtensionTypeAnnotationsRegistry` to provide a readable serialization.